packages/cli/__tests__/zosfiles/__system__/upload/dtu/cli.dir.upload.dtu.system.test.ts

```diff
@@ -215,16 +215,27 @@ describe("Upload directory to USS", () => {
             expect(response.stdout.toString()).toContain("Directory uploaded successfully.");
         });
 
-        it("should give error when upload local directory to USS directory in default ascii if it contains also binary files", async () => {
+        it("should not give error when upload local directory to USS directory in default ascii if it contains also binary files", async () => {
+            const localFileLocation = path.join(TEST_ENVIRONMENT.workingDir, "bin_file.pax");
             const localDirName = path.join(__dirname, "__data__", "command_upload_dtu_dir");
             const shellScript = path.join(__dirname, "__scripts__", "command", "command_upload_dtu.sh");
             const response = runCliScript(shellScript, TEST_ENVIRONMENT,
                 [
                     localDirName,
                     ussname
                 ]);
-            expect(response.stderr.toString()).toContain("Rest API failure with HTTP(S) status 500");
-            expect(response.status).toBe(1);
+
+
+            // const downloadResponse = await Download.ussFile(REAL_SESSION, path.posix.join(ussname, "bin_file.pax"), { file: localFileLocation });
+            // expect(downloadResponse.success).toBe(true);
+            // // fs.readFileSync(localFileLocation).toString(); returns an empty buffer ???? Getting the file directly seems to work consistently
+            // const downloadedFileContents = fs.readFileSync(localFileLocation).toString();
+
+            const downloadedFileContents = (await Get.USSFile(REAL_SESSION, path.posix.join(ussname, "bin_file.pax"), {binary: false})).toString();
+            expect(downloadedFileContents).toContain("00000000125");
+            expect(downloadedFileContents).toContain("13424013123");
+            expect(response.stderr.toString()).not.toContain("Rest API failure with HTTP(S) status 500");
+            expect(response.status).not.toBe(1);
```

Comment:
Curious if this change was intended, and whether it is warranted by a change that was made in z/OSMF?

Reply:
It seems like z/OSMF auto-converts binary to text data now when downloading a file instead of blowing up. Thanks @awharn for explaining 🙂